File: ios/try_flutter_plugin/Sources/try_flutter_plugin/TryFlutterPlugin.swift
import Flutter
import TryKMP
import UIKit

public class TryFlutterPlugin: NSObject, FlutterPlugin, TryKmpHostApi {
    public static func register(with registrar: FlutterPluginRegistrar) {
        TryKmpHostApiSetup.setUp(binaryMessenger: registrar.messenger(), api: TryFlutterPlugin())
    }
    
    
    private let model = TryKmp()
    
    
    func searchGitHubRepo(query: String, completion: @escaping (Result<GitHubRepoDto, Error>) -> Void){
        model.searchGitHubRepo(query: query) { result, error in
            if let result = result {
                let mapped = GitHubRepoDto(
                    totalCount: Int64(result.totalCount),
                    incompleteResults: result.incompleteResults,
                    items: result.items.map {
                        GitHubRepoItemDto(
                            fullName: $0.fullName,
                            description: $0.description,
                            url: $0.url,
                            updatedAt: $0.updatedAt.ISO8601Format(),
                            language: $0.language
                        )
                    }
                )
                completion(.success(mapped))
            }
            completion(.failure(error ?? PigeonError(code: "", message: nil, details: nil)))
        }
    }
    
    func time() throws -> String {
        return model.time(suffix: "")
    }
}

File: android/src/main/kotlin/io/github/tshion/try_flutter_plugin/TryFlutterPlugin.kt
package io.github.tshion.try_flutter_plugin

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.FlutterPlugin.FlutterPluginBinding
import io.github.tshion.trykmp.TryKmp
import kotlinx.coroutines.runBlocking

/** TryFlutterPlugin */
class TryFlutterPlugin : FlutterPlugin, TryKmpHostApi {

    private val model by lazy { TryKmp() }


    override fun onAttachedToEngine(binding: FlutterPluginBinding) {
        TryKmpHostApi.setUp(binding.binaryMessenger, this)
    }

    override fun onDetachedFromEngine(binding: FlutterPluginBinding) {
        TryKmpHostApi.setUp(binding.binaryMessenger, null)
    }


    override fun time(): String {
        return model.time()
    }

    override fun searchGitHubRepo(
        query: String,
        callback: (Result<GitHubRepoDto>) -> Unit,
    ) {
        runBlocking {
            try {
                val result = model.searchGitHubRepo(query)
                result.let { native ->
                    GitHubRepoDto(
                        native.totalCount.toLong(),
                        native.incompleteResults,
                        native.items.map {
                            GitHubRepoItemDto(
                                it.fullName,
                                it.description,
                                it.url,
                                it.updatedAt.toString(),
                                it.language,
                            )
                        },
                    )
                }.also { callback(Result.success(it)) }
            } catch (e: Exception) {
                callback(Result.failure(e))
            }
        }
    }
}
